mantidproject/mantid
# Creating OSX installer from conda environment

**Description of work.**
Builds upon #32903. This PR creates a package installer from our conda environment. It uses the standalone install in #32903 and CPack to create a .pkg installer.




**To test:**
NOTE: An installer created with this PR https://www.dropbox.com/s/bresxfji83d73n8/mantidunstable-6.2.20211110.923-HighSierra.pkg

Build mantid:
```
cmake .. --preset=osx -DENABLE_CPACK=ON -DCMAKE_INSTALL_PREFIX=../../my_install_folder/mantid -DCMAKE_
BUILD_TYPE=Release
ninja
```
Package mantid

```
cmake --install . --component Runtime
cpack -C Release
```
This should create a .pkg in the build folder. Now install this. You will get a `MantidUnstableWorkbench` app in /Applications. 

- Test this package out. 
- It should be fully functional
- It should contain everything necessary to run workbench


Just my own comment. This went in circles a bit between trying to use BundleUtilities from CMake. I found out that they didn't recommend this for packages which had executables located in more than one location https://cmake.org/pipermail/cmake/2011-February/042428.html. We are shipping python in a folder outside of MacOS (non flat structure). CMake does no expect this/ bundle utilities doesn't work in this case. This is because BundleUitilites will replace all the rpaths pointing to the Framework libs with:


@exectuable_path/../Framework. 

If all the executables are located in MacOS then all then executable_path/../Frameworks always correctly resolves. If however, we have a python executable in Contents/Resources/mantid_python/bin - the @executable_path will resolve to Resources/Frameworks, a non exsistent path. The solution would be to try and get the python executable into the MacOS - or rewrite BundleUtilities to suit our needs.

 I chose the latter, where I just use BundleUtiltieis to verify that all the dependencies I need for the MantidExecutable and libraries are contained in the package somewhere. To do this I've written a custom Bundle file based on CMake. 




Fixes #33024 . 


This does not require release notes



---


#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
